### PR TITLE
Adding notes on the practice of using `module reset`

### LINKED ISSUE #524: New good practice for environment management and modules
### What would you like to see added?

Clearly document the need for `module reset` at the start of any sbatch script.

1. What it causes to happen
2. What it prevents from happening
3. Why it is a best-practice

Put this in...

- https://docs.rc.uab.edu/cheaha/software/modules/ (primary location)
- https://docs.rc.uab.edu/cheaha/slurm/submitting_jobs/ (subsection + link)
- https://docs.rc.uab.edu/cheaha/slurm/practical_sbatch/ (admonition + link)
- https://docs.rc.uab.edu/cheaha/slurm/gpu/#scheduling-gpus (admonition + link)
- https://docs.rc.uab.edu/cheaha/open_ondemand/ood_interactive/ (break out a subsection for "Environment Setup" box, put admonition + link at that spot)

Add an ask.ci FAQ about...

- Why do I see `The following have been reloaded with a version change?`

    May indicate a dependency conflict caused by loading a module with a dependency that replaces another loaded module's dependency. As an example, suppose you `module load A` which depends on `module D/2.0`. Then you `module load B` which depends on `module D/1.1`. You will get the warning and `module D/2.0` will be unloaded and replaced by `module D/1.1`.

    Proceeding despite the conflict may result in unexpected behavior, including fatal errors such as segmentation faults (link ask.ci). If you encounter such a warning in your workflow, please see if Anaconda has up-to-date versions of the software you need (link). If so, try building an Anaconda environment (link). If you still need assistance please contact us (link).

    Best practice is to use `module reset` prior to loading modules (link) at the terminal and in `sbatch` scripts. Do not load modules in your `.bashrc` file, and only use the defaults or `module reset`. The goal is to give each job its own separate, clean, reproducible environment to work in.

Link from the `segfault` ask.ci FAQ to the proposed one.

### PR DESCRIPTION
**Pull request by bdu-birhanu**
## Overview
Notes on the practice of using `module reset` command at the start of sbatch scripts.

## Proposed Changes
- Adding notes on the practice of `module reset' in [modules and applications](https://docs.rc.uab.edu/cheaha/software/modules/) page.
- crosslink at: [Job submission](https://docs.rc.uab.edu/cheaha/slurm/submitting_jobs/), [Practical sbatch](https://docs.rc.uab.edu/cheaha/slurm/practical_sbatch/),  [GPU](https://docs.rc.uab.edu/cheaha/slurm/gpu/#scheduling-gpus),  and [OOD](https://docs.rc.uab.edu/cheaha/open_ondemand/ood_layout/) pages.

## Related Issues
Fixes #524

### CODE REVIEW

**Review by mdefende — CHANGES_REQUESTED** [review]
Change py designation in code blocks back to bash where bash is being used

File: docs/cheaha/software/modules.md [hunk 1]
```diff
@@ -98,7 +98,18 @@ To reduce unexpected behavior and/or to get rid of Lmod errors,
     TopHat/2.1.1-foss-2016a
     ```
 
-2. Before loading modules in a shell/bash/sbatch script, use a clean shell by using `module reset` at the beginning to restore to default system settings. Using `module reset` before loading modules separates what software is loaded in the working shell from the software loaded in the script shell. Be aware that forked processes (like scripts) and Slurm commands inherit the environment variables of the working shell, including loaded modules. Here is an example that shows module conflict between cuda11.8 and cuda11.4 versions that may lead to unexpected behavior or an erroneous output.
+2. Be selective and only load a specific module version that you need for your current workflow.Loading unnecessary modules can lead to conflicts and inefficiencies.
```
Comment (mdefende) [inline docs/cheaha/software/modules.md:101]: Add space before Loading

**Comment by bdu-birhanu** [comment]
> Change py designation in code blocks back to bash where bash is being used

Just fixed.

File: docs/cheaha/software/modules.md (hunk 1 above)
Comment (bdu-birhanu) [reply docs/cheaha/software/modules.md]: Fixed

**Review by wwarriner — APPROVED** [review]
Looks good!

Y'all may need to approve the changes or dismiss the review to proceed.

**Review by mdefende — APPROVED** [review]

**Review by mdefende — APPROVED** [review]

**Pull request merged**